### PR TITLE
Add AddressPool validation webhook

### PR DESCRIPTION
**Pull request by sabinaaledort**
Add a validation webhook to avoid duplicate IP addresses in AddressPools.

In case of a duplicate IP range when applying an AddressPool, the request will be denied with an error message.

### CODE REVIEW

File: manifests/4.9/metallb-operator.v4.9.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -384,3 +399,25 @@ spec:
   provider:
     name: Community
   version: 4.9.0
+  webhookdefinitions:
```
[the remark this answers is not in the document]
Comment (sabinaaledort) [reply manifests/4.9/metallb-operator.v4.9.0.clusterserviceversion.yaml:402]: there is no clientConfig in the webhook definition:
https://github.com/operator-framework/api/blob/dd58de2b37096fb5fcd8b5538cf58cfc811b113a/pkg/operators/v1alpha1/clusterserviceversion_types.go#L175

[the remark this answers is not in the document]
Comment (SchSeba) [reply manifests/4.9/metallb-operator.v4.9.0.clusterserviceversion.yaml:402]: Just a quick note here :)

The webhook configuration was needed to be deployed by the operator itself before OLM API expose the webhook configuration that is why both multus and sriov need it.

Now only multus will need it as it's not installed by OLM but in the sriov operator we should be able to drop it, also dropping the RBAC's from the operator for better security leave the OLM as the only one creating the webhooks.

OLM also handles the clientConfig adding the CA generation and renewal. that section is not exposed in the OLM CSV CRD.

[the remark this answers is not in the document]
Comment (sabinaaledort) [reply manifests/4.9/metallb-operator.v4.9.0.clusterserviceversion.yaml:402]: just deployed the operator with OLM on OCP. it seems that indeed the clientConfig and the CA were set by OLM.

```
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  generateName: addresspoolvalidationwebhook.metallb.io-
  generation: 1
  labels:
    olm.owner: metallb-operator.v0.1.0
    olm.owner.kind: ClusterServiceVersion
    olm.owner.namespace: metallb-system
    olm.webhook-description-generate-name: addresspoolvalidationwebhook.metallb.io
    olm.webhook-description-hash: ffd6b5574
webhooks:
- admissionReviewVersions:
  - v1
  - vibeta1
  clientConfig:
    caBundle: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJoekNDQVMyZ0F3SUJBZ0lJV1E2OTBnanFPSEl3Q2dZSUtvWkl6ajBFQXdJd0dERVdNQlFHQTFVRUNoTU4KVW1Wa0lFaGhkQ3dnU1c1akxqQWVGdzB5TVRFd01UY3hNekEzTWpoYUZ3MHlNekV3TVRjeE16QTNNamhhTUJneApGakFVQmdOVkJBb1REVkpsWkNCSVlYUXNJRWx1WXk0d1dUQVRCZ2NxaGtqT1BRSUJCZ2dxaGtqT1BRTUJCd05DCkFBU25Od3JEY2RUc2kwWUluS1NJbmpqbUx1VjM4V0VMYkt0dzVtbmUrUk9abGFQZFEvbnFoYjRFOXpYcE5uQlYKWjBDZk02bU9tWWR1U2swQVpocWZYM1VRbzJFd1h6QU9CZ05WSFE4QkFmOEVCQU1DQW9Rd0hRWURWUjBsQkJZdwpGQVlJS3dZQkJRVUhBd0lHQ0NzR0FRVUZCd01CTUE4R0ExVWRFd0VCL3dRRk1BTUJBZjh3SFFZRFZSME9CQllFCkZLcXpJNnUxM2MrQWVkdWozZ0pNbkE0R2t0ajFNQW9HQ0NxR1NNNDlCQU1DQTBnQU1FVUNJQ2dLd3lUeWU4aEgKS09PRkQ5Y25qRm5kRWhDR0ZPODNTWVdwN3FtQXg4WElBaUVBL29jcmYvVlBhU2ZNN2VjdSs5UUVtbzgvc2I3cQpiYVZGWHE4OFovK3NLSzQ9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
    service:
      name: metallb-operator-controller-manager-service
      namespace: metallb-system
      path: /validate-metallb-io-v1alpha1-addresspool
      port: 443
  failurePolicy: Fail
  matchPolicy: Equivalent
  name: addresspoolvalidationwebhook.metallb.io
  namespaceSelector:
    matchLabels:
      olm.operatorgroup.uid/e1a17c54-d4a4-4507-863c-1ca407223978: ""
```

File: Makefile [hunk 2]
```diff
@@ -43,6 +43,8 @@ CERT_MANAGER_URL=https://github.com/jetstack/cert-manager/releases
 
 TESTS_REPORTS_PATH ?= /tmp/test_e2e_logs/
 
+IS_OPENSHIFT ?= 0
```
[the remark this answers is not in the document]
Comment (sabinaaledort) [reply Makefile:46]: ok let's do it u/s first
https://github.com/metallb/metallb-operator/pull/102

**Comment by sabinaaledort** [comment]
was added in https://github.com/openshift/metallb-operator/pull/37

**Pull request closed without merging**